File: 백준/Gold/1107. 리모컨/리모컨.py
def make_num(idx, length, num):
    if idx == length:
        nums.append(num)
        return
    for i in range(10):
        if i not in broken:
            make_num(idx+1, length, num*10+i)

n = int(input())
m = int(input())
if m != 0:
    broken = list(map(int, input().split()))
else:
    broken = []
k = 1
tmp = n
while tmp >= 10:
    k += 1
    tmp //= 10
i = 0
min_val = 500000
while i <= k+1:
    if i == 0:
        if min_val > abs(n-100):
            min_val = abs(n-100)
    else:
        nums = []
        make_num(0, i, 0)
        for num in nums:
            if min_val > i+abs(n-num):
                min_val = i+abs(n-num)
    i += 1
print(min_val)

File: 백준/Gold/1107. 리모컨/리모컨.cc
#include <iostream>
#include <string>
#include <vector>
#define FIO ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);

using namespace std;

vector<int> nums;
bool broken[10] = {false, };

void makeNum(int idx, int len, int num) {
	if (idx == len) {
		nums.push_back(num);
		return;
	}
	for (int i = 0; i < 10; i++)
		if (!broken[i]) makeNum(idx + 1, len, num * 10 + i);
}

int main() {
	FIO;

	int n, m, tmp;
	cin >> n >> m;
	for (int i = 0; i < m; i++) {
		cin >> tmp;
		broken[tmp] = true;
	}
	int len = to_string(n).length();
	int minVal = abs(n - 100);
	for (int i = 1; i <= len+1; i++) {
		nums.clear();
		makeNum(0, i, 0);
		while (!nums.empty()) {
			int num = i + abs(n - nums.back());
			nums.pop_back();
			if (minVal > num) minVal = num;
		}
	}
	cout << minVal;

	return 0;
}